addons/forum-post-countdown/userscript.js

```diff
@@ -1,10 +1,13 @@
+import { onPost, onNonSuccessfulPost } from "../auto-follow-topics/module.js";
+
 export default async function ({ addon, msg }) {
   const submitButton = document.querySelector("#djangobbwrap .form-submit [type=submit]");
 
-  submitButton.addEventListener("click", () => {
-    if (!localStorage.getItem("sa-forum-post-countdown")) {
-      localStorage.setItem("sa-forum-post-countdown", Date.now());
-    }
+  onPost(() => {
+    localStorage.setItem("sa-forum-post-countdown", Date.now());
+  });
+  onNonSuccessfulPost(() => {
+    localStorage.removeItem("sa-forum-post-countdown");
   });
```

Comment:
```suggestion
  onSuccessfulPost(() => {
    localStorage.setItem("sa-forum-post-countdown", Date.now());
  });
```
This would make the most sense and is probably what the person who reported #7269 wanted. There's no reason to remove the cooldown when a post fails.